### DISCUSSION TITLE
Changing text colors?

### DISCUSSION THREAD

**Discussion by LedLoaf** [gabime/spdlog#2706] [Q&A] [answered]
I don't know why this has become so difficult for me to look up.

I have a core logger and a client logger. All I'm trying to do is change the color of the text for one of the levels. Debug in this case.
I've read on here a SetTextColor option and I don't see that in the current version. I also saw someone mention using %^ color here %$ and that doesn't seem to do anything for me either, unless I'm not specificying the color correctly.

Any help would be appreciated. I feel like color options/example snippets use to be on the front page?

**Comment by gabime** [comment]
Please see https://github.com/gabime/spdlog/wiki/0.-FAQ#colors-do-not-appear-when-using-custom-format

**Reply by LedLoaf** [reply]
Hey thanks, I saw that part. So what is [%l] ? 
How do you actually specify the color?

**Reply by tt4g** [reply]
`%l` is a pattern flag.
3. Custom formatting: https://github.com/gabime/spdlog/wiki/3.-Custom-formatting

Note that some of the Windows consoles, e.g. the debugger console, do not support color.

**Comment by gabime** [comment, marked as answer]
The color is determined automatically by spdlog according to the log level.
To change the mapping of levels to colors use`void set_color(level::level_enum level, std::uint16_t color))` https://github.com/gabime/spdlog/blob/0ca574ae168820da0268b3ec7607ca7b33024d05/include/spdlog/sinks/wincolor_sink.h#L34

**Reply by LedLoaf** [reply]
Thanks!
